tests/v4l2decode.cpp

```diff
@@ -590,15 +590,21 @@ int main(int argc, char** argv)
     memset(&format, 0, sizeof(format));
     format.type = V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE;
     while (1) {
-        if (SIMULATE_V4L2_OP(Ioctl)(fd, VIDIOC_G_FMT, &format) != 0) {
-            if (errno != EINVAL) {
-                // EINVAL means we haven't seen sufficient stream to decode the format.
-                INFO("ioctl() failed: VIDIOC_G_FMT, haven't get video resolution during start yet, waiting");
-            }
-        } else {
+        ioctlRet = SIMULATE_V4L2_OP(Ioctl)(fd, VIDIOC_G_FMT, &format) ;
+        if (ioctlRet == 0)
             break;
+
+        #ifdef YUNOS
+        if (ioctlRet == EAGAIN) {
+        #else
+        if (errno == EINVAL) {
+            // EINVAL means we haven't seen sufficient stream to decode the format.
+        #endif
+            INFO("ioctl() failed: VIDIOC_G_FMT, haven't get video resolution during start yet, waiting");
+        } else {
+            WARNING("ioctl fail to get VIDIOC_G_FMT");
         }
-        usleep(50);
+        usleep(5000);
```

Comment:
Suggest using other means such as a callback to indicate when resolution info is available, rather than hardcoding the wait time.  This is not efficient especially because 'usleep' is called within an infinite 'while' loop.  So every time you run thru the loop, it blocks for 5 ms.